enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,498 @@
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  creationTimestamp: null
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+[...]
+  endpointPublishingStrategy:
+    loadBalancer:
+      providerParameters:
+        aws:
+          networkLoadBalancer:
+            managedSecurityGroup: true   <-- new field
+          type: NLB
+        type: AWS
+      scope: External
+    type: LoadBalancerService
+[...]
+```
+- Cluster Ingress Operator (CIO):
+  - CIO will create the Service instance for the default router. If the `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.managedSecurityGroup` is `true`, CIO will include the following annotation in the Service manifest:
+```yaml
+# Manifest for Service XYZ is created with annotations:
+apiVersion: v1
+kind: Service
+metadata:
+  name: echoserver
+  namespace: mrbraga
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: nlb
+    service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
+    service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true" <-- new annotation
+[...]
+```
+- Cloud Controller Manager (CCM):
+  - CCM validates the annotation to manage SG on NLB, creates the SG and rules, and pass the SG ID to LB creation.
+  - The annotation `service.beta.kubernetes.io/aws-load-balancer-managed-security-group` (new) msut be set to `true`, then creates the SG with required rules for ingress (based on listeners) and egress (based on service and health check ports).
+  - CCM LB controller manages the SG lifecycle (controllers may exists in CLB).
+
+  - The CCM's service controller will watch for Service creations and updates.
+  - When it encounters a Service with the annotation `service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true"` and `service.beta.kubernetes.io/aws-load-balancer-type: nlb`, the CCM will:
+    - Create a new AWS Security Group for the NLB. The name should follow a convention like `k8s-elb-a<generated-name-from-service-uid>`.
+    - Configure Ingress rules in the Security Group to allow traffic on the ports defined in the Service's `spec.ports`. The source for these rules will be determined by the `service.beta.kubernetes.io/load-balancer-source-ranges` annotation on the Service (if present, otherwise default to allowing from all IPs).
+    - Configure Egress rules in the Security Group to allow traffic to the backend pods on the targetPort specified in the Service's `spec.ports` and the health check port. Initially, this should be restricted to the cluster's VPC CIDR or the specific CIDRs of the worker nodes.
+    - When creating the NLB using the AWS ELBv2 API, the CCM will include the ID of the newly created Security Group in the `SecurityGroups` parameter of the `CreateLoadBalancerInput.`
+  - When the Service is deleted, the CCM will also delete the associated Security Group, ensuring proper cleanup.
```

Comment:
what happens if this annotation is added after the Service is created? (ie what happens on update)

Reply:
I am working on it, ensuring I will follow the current state of CCM along side the ALBC to correctly document it. Thanks for raising that question.

Reply:
We need to be able to answer these questions for upstream, but downstream we could prevent those transitions with VAP